app/controllers/StudyController.java

```diff
@@ -54,7 +50,7 @@ public Result updateStudyForResearcher() throws Exception {
         UserSession session = getAuthenticatedResearcherSession();
         StudyIdentifier studyId = session.getStudyIdentifier();
 
-        Study studyUpdate = DynamoStudy.fromJson(requestToJSON(request()));
+        Study studyUpdate = parseJson(request(), DynamoStudy.class);
```

Comment:
You can say parseJson(request(), Study.class), and the @JsonDeserialize(as=DynamoStudy.class) annotation in study will make Jackson do the right thing.